src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -924,6 +932,47 @@ bool OutputStateMachineEngine::_ParseHyperlink(const std::wstring_view string,
     return false;
 }
 
+// Routine Description:
+// - OSC 9 ; 4 ; state ; progress ST
+//       state: the state to set the taskbar progress indicator to
+//       progress: the progress value to set the taskbar progress indicator with
+// - Parses out the 'state' and 'progress' parameters from the OSC string so
+//   that we can use them to set the taskbar progress indicator
+// - This follows the ConEmu style, more details here:
+//   https://conemu.github.io/en/AnsiEscapeCodes.html#ConEmu_specific_OSC
+// Arguments:
+// - string: the string to parse
+// - state: where to store the state value once we parse it
+// - progress: where to store the progress value once we parse it
+// Return Value:
+// - true if we successfully parsed the string, false otherwise
+bool OutputStateMachineEngine::_GetTaskbarProgress(const std::wstring_view string,
+                                                   size_t& state,
+                                                   size_t& progress) const
+{
+    const auto parts = Utils::SplitString(string, L';');
+    if (parts.size() < 1 || std::stoi(til::u16u8(parts.at(0))) != 4)

```

Comment:
Here as @DHowett would probably do, I'll suggest use `til::at(parts, 0)` is the index is guaranteed to be valid.

Reply:
Also the hand-written `Utils::StringToUint` should be theoretically faster than `std::stoi(til::u16u8())`. Plus it does not throw exceptions.

Reply:
Very handy tips, thank you!